### ISSUE TITLE
Automated test definition generation hiding simulation errors

### ISSUE THREAD

**Issue by AnHeuermann** [pstelzig/MoPyRegtest#36]
# Problem description

The command line interface fails to generate test definition files if the Modelica model can't be translated/simulated.

**To Reproduce**

The automated test definition generation hides simulation errors for `MyLibrary.M`:

```modelica
package MyLibrary "Package"
  model M "Model"
    Real x(fixed=true, start=0);
  equation
    der(x) = -0.5*x + sqrt(0.5 - time);  // Breaks simulation at time=0.5
  end M;
end MyLibrary;
```

```bash
> mopyregtest generate .CI/mopyregtest/ "TestM" . MyLibrary.M
Simulating model MyLibrary.M using the simulation tools: omc
Using simulation tool omc
```

The resulting reference file `.CI/mopyregtest/references/MyLibrary.M_res.csv` is missing half of the simulation, but `mopyregtest` exited without any errors.

**System information (please complete the following information):**

 - OS: Ubuntu 22.04.3 LTS
 - OpenModelica: v1.23.0-dev-203-g5904d1ea84-cmake
 - Python: v3.10.12
 - MoPyRegtest: v0.5.0

# Desired behavior

Return an error when the simulation for reference file generation failed.
Print or log the error message returned by OpenModelica.

So something like:

```
> mopyregtest generate .CI/mopyregtest/ "TestM" . MyLibrary.M
Simulating model MyLibrary.M using the simulation tools: omc
Using simulation tool omc

Failed to simulate model MyLibrary.M

LOG_STDOUT        | warning | A Modelica assert prevents the integrator to continue. For more information use -lv LOG_SOLVER
LOG_STDOUT        | warning | can't continue. time = 0.500000
LOG_STDOUT        | info    | model terminate | Integrator failed. | Simulation terminated at time 0.5

The full log can be viewed at some/path/MyLibrary.M.log
# exit code 1
```

**Comment by pstelzig** [comment]
To avoid having a regression test fail on running, or a test generation fail on generating a valid reference file, I did two things: 
1. In `RegressionTest._run_model` I now delete any old simulation binary or old simulation result to avoid using either of them by error in the regression test
2. At the very end of `RegressionTest._run_model` I check if both a simulation binary and simulation result file have been produced. Since I deleted any existing ones before, any existing at that point must be new. If either one is missing, I raise an AssertionError. 

See commit 9811d53a6943fcde5153f7d6499fb2113d2e58d2 and the unit tests for that e6ce5b4e8c84a730a33afc488ed15196a5fbdf00. 

Thanks @AnHeuermann for the minimum working example to reproduce the issue!

**Comment by pstelzig** [comment]
I experimented with looking for error or fatal messages in the log output produced by the simulation binary, see 3ccff5c415b7f431b1b160a45016040f6d8a30de. However, this is quite difficult to judge. Issues are
- Sometimes log messages are declared error at one point in the log, then they appear as warnings at some other point. 
- Sometimes modelers willfully accept errors from a solver, as long as the desired output is produced

Therefore, I removed the parsing of simulation output in favor of the solution described in the previous comment, see 9811d53a6943fcde5153f7d6499fb2113d2e58d2. In short words, I check if all necessary artifacts for running regression tests have been produced, i.e.
1. A simulation binary has been produced that
2. produces a simulation result file to run the result comparison on.
